seek.h

```diff
@@ -1,14 +1,35 @@
 #ifndef SEEK_H
 #define SEEK_H
 
-typedef unsigned long seek_pos;
+#include <sys/types.h>
+#include <unistd.h>
 
-extern seek_pos seek_cur();
+typedef off_t seek_pos;
 
-extern int seek_set();
-extern int seek_end();
+static inline seek_pos seek_cur(int fd)
+{
+  return lseek(fd, 0, SEEK_CUR);
+}
 
-extern int seek_trunc();
+static inline int seek_set(int fd, seek_pos pos)
+{
+  if (lseek(fd, pos, SEEK_SET) == -1)
+    return -1;
+  return 0;
+}
+
+
+static inline int seek_end(int fd)
+{
+  if (lseek(fd, 0, SEEK_END) == -1)
+    return -1;
+  return 0;
+}
+
+static inline int seek_trunc(int fd, seek_pos pos)
```

Comment:
Out of curiosity, why not `#define seek_trunc(fd, pos) (ftruncate(fd, pos))`, at least for this one with no branching logic?

Reply:
sure, would be consistent with other changes.